### PR TITLE
[Semi-Auto] Add transpose spmd rule

### PR DESCRIPTION
**Pull request by pkuzyc**
### PR types
New features

### PR changes
Others

### Description
Pcard-70448
1. add forward spmd rule of transpose for auto parallel. 
2. fix some typo and bugs in common functions.

### CODE REVIEW

File: paddle/fluid/distributed/auto_parallel/spmd_rules/transpose_spmd_rule.cc [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+  // initialize output dist_attr's process_mesh, batch_dim and dynamic dims with
+  // input dist_attr.
+  TensorDistAttr output_dist_attr =
+      CopyTensorDistAttrForOutput(input_specs[0].dist_attr());
+  output_dist_attr.set_dims_mapping(output_dims_mapping);
+
+  // Step2.3  handle input tensor partial (TODO)
+  VLOG(4) << "TransposeSPMDRule InferForward:";
+  for (int64_t i = 0; i < ninputs; i++) {
```
Comment (JZ-LIANG) [inline paddle/fluid/distributed/auto_parallel/spmd_rules/transpose_spmd_rule.cc:77]: log should  print attr that would effect spmd inference, like perm here.

**Review by JZ-LIANG — DISMISSED** [review]
LGTM

File: paddle/fluid/distributed/auto_parallel/spmd_rules/transpose_spmd_rule.cc (hunk 1 above)
Comment (pkuzyc) [reply paddle/fluid/distributed/auto_parallel/spmd_rules/transpose_spmd_rule.cc:77]: Done

**Review by JZ-LIANG — APPROVED** [review]
LGTM

**Pull request merged**